graphql-dgs-codegen-shared-core/src/main/kotlin/com/netflix/graphql/dgs/client/codegen/GraphQLMultiQueryRequest.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ *
+ *  Copyright 2020 Netflix, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.netflix.graphql.dgs.client.codegen
+
+import graphql.language.*
+
+class GraphQLMultiQueryRequest(
+    private val requests: List<GraphQLQueryRequest>
+) {
+
+    fun serialize(): String {
+        if (requests.isEmpty()) throw AssertionError("Request must have at least one query")
+
+        if (requests.size == 1) {
+            return requests[0].serialize()
+        }
+
+        return multiQuery()
+    }
+
+    private fun multiQuery(): String {
+        val operationDef = OperationDefinition.newOperationDefinition()
+        requests[0].query.name?.let { operationDef.name(it) }
+        requests[0].query.getOperationType()?.let { operationDef.operation(OperationDefinition.Operation.valueOf(it.uppercase())) }
+
+        val queryType = requests[0].query.getOperationType().toString()
+        val variableDefinitions = mutableListOf<VariableDefinition>()
+        val selectionList: MutableList<Field.Builder> = mutableListOf();
+
+        for (request in this.requests) {
+            val query = request.query
+            if (!query.getOperationType().equals(queryType) || queryType == OperationDefinition.Operation.SUBSCRIPTION.name)
```

Comment:
What is the check for SUBSCRIPTION doing in this line?

Reply:
From my understanding of[ this](http://spec.graphql.org/June2018/#sec-Single-root-field), only one subscription is allowed per request. So I check to make sure they are not sending a subscription via the new GraphQLMultiQueryRequest object.

Reply:
I see, that makes sense. Thanks for clarifying. Would be good to add that as a comment, if possible.

Reply:
Just pushed the comment. Great suggestion!